server/cluster/cluster.go

```diff
@@ -0,0 +1,338 @@
+				if ok {
+					tables[tableID] = &Table{
+						schema: c.metaData.schemaMap[schemaName],
+						meta:   table,
+					}
+				}
+			}
+		}
+		// todo: assert shardID
+		// todo: check shard not found by shardID
+		shardMetaList := c.metaData.shardMap[shardID]
+		var nodes []*clusterpb.Node
+		for _, shardMeta := range shardMetaList {
+			nodes = append(nodes, c.metaData.nodeMap[shardMeta.Node])
+		}
+		c.shards[shardID] = &Shard{
+			meta:    c.metaData.shardMap[shardID],
+			nodes:   nodes,
+			tables:  tables,
+			version: 0,
+		}
+	}
+
+	for shardID, shards := range c.metaData.shardMap {
+		for _, shard := range shards {
+			if _, ok := c.nodes[shard.GetNode()]; ok {
+				c.nodes[shard.GetNode()] = append(c.nodes[shard.GetNode()], shardID)
+			} else {
+				c.nodes[shard.GetNode()] = []uint32{shardID}
+			}
+		}
+	}
+
+	return nil
+}
+
+func (c *Cluster) updateSchemaCacheLocked(schemaPb *clusterpb.Schema) *Schema {
+	schema := &Schema{meta: schemaPb}
+	c.schemas[schemaPb.GetName()] = schema
+	return schema
+}
+
+func (c *Cluster) updateTableCacheLocked(schema *Schema, tablePb *clusterpb.Table) *Table {
+	table := &Table{meta: tablePb, schema: schema.meta}
+	schema.tableMap[tablePb.GetName()] = table
+	c.shards[tablePb.GetShardId()].tables[table.getID()] = table
+	return table
+}
+
+func (c *Cluster) GetTables(ctx context.Context, shardIDs []uint32, node string) (map[uint32]*ShardTablesWithRole, error) {
+	shardTables := make(map[uint32]*ShardTablesWithRole, len(shardIDs))
+	for _, shardID := range shardIDs {
+		shardTable, ok := c.shards[shardID]
+		if !ok {
+			return nil, ErrShardNotFound.WithCausef("shardID", shardID)
+		}
+
+		shardRole := FOLLOWER
+		for i, n := range shardTable.nodes {
+			if node == n.GetName() {
+				if shardTable.meta[i].ShardRole == clusterpb.ShardRole_LEADER {
+					shardRole = LEADER
+				} else {
+					shardRole = FOLLOWER
+				}
+				break
+			}
+		}
+		tables := make([]*Table, len(shardTable.tables))
+		for _, table := range shardTable.tables {
+			tables = append(tables, table)
+		}
+		shardTables[shardID] = &ShardTablesWithRole{shardRole: shardRole, tables: tables, version: shardTable.version}
+	}
+
+	return shardTables, nil
+}
+
+func (c *Cluster) DropTable(ctx context.Context, schemaName, tableName string, tableID uint64) error {
+	schema, exists := c.getSchema(schemaName)
+	if exists {
+		return ErrSchemaNotFound.WithCausef("schemaName:%s", schemaName)
+	}
+	if err := c.storage.DeleteTables(ctx, c.clusterID, schema.GetID(), []uint64{tableID}); err != nil {
+		return errors.Wrapf(err, "cluster DropTable, "+
+			"clusterID:%d, schema:%v, tableID:%d", c.clusterID, schema, tableID)
+	}
+	c.Lock()
+	defer c.Unlock()
+	schema.dropTableLocked(tableName)
+	for _, shard := range c.shards {
+		shard.dropTableLocked(tableID)
+	}
+	return nil
+}
+
+func (c *Cluster) CreateSchema(ctx context.Context, schemaName string, schemaID uint32) (*Schema, error) {
+	c.Lock()
+	defer c.Unlock()
+	// check if exists
```

Comment:
Shall we put comments as full sentences?
Ref: https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences